### PR TITLE
Move Task Instance to reference

### LINKED ISSUE #174: Add TaskInstance docs
We mainly want to move it from "Basic Usage" to "Reference"

### PR DESCRIPTION
**Pull request by beerinho**
This PR moves the TaskInstance docs to "reference" and refers to its new home in the "Basic Usage" section

Closes #174

### CODE REVIEW

File: apps/docs/src/content/docs/getting-started/usage.mdx [hunk 1]
```diff
@@ -78,46 +78,7 @@ type Task = {
...
-	 */
-	isSuccessful: boolean;
-
-	/**
-	 * If the task instance completed successfully, this will be the return value.
-	 */
-	value?: TReturn;
-};
-```
+To make it easier to reason about, every type of task modifier utilizes the same underlying structure. You can find more detail about the structure of the `TaskInstance` in the [reference docs](/reference/task-instance).
```
Comment (paoloricciuti) [inline apps/docs/src/content/docs/getting-started/usage.mdx:81]: I think we can use the Link component from Astro to have nice link boxes

Comment (beerinho) [reply apps/docs/src/content/docs/getting-started/usage.mdx:81]: Can you point me towards this component, I can't see it anywhere. Also, we aren't using the Link component in any of the other reference docs - https://github.com/mainmatter/sheepdog/blob/b8bc4692bc4a7dd3ceeb22e0853e7dba5a13ffe8/apps/docs/src/content/docs/reference/default.mdx?plain=1#L181

Comment (paoloricciuti) [reply apps/docs/src/content/docs/getting-started/usage.mdx:81]: It's the `LinkCard` component, you can see usage in various places.

Comment (beerinho) [reply apps/docs/src/content/docs/getting-started/usage.mdx:81]: I'm not a big fan of how this looks to be honest. What do you think?
CleanShot 2024-09-27 at 13 20 07

Comment (paoloricciuti) [reply apps/docs/src/content/docs/getting-started/usage.mdx:81]: Mmm yeah I think you are right

**Review by paoloricciuti — APPROVED** [review]

**Pull request merged**